### PR TITLE
Adding support for ME007-ULS narrow FoV ultrasonic distance sensor

### PR DESCRIPTION
**Pull request by mathias-buder**
## Description:
I'm currently working on a system to measure the level of water within a large fresh water tank ([water level measurement system](https://github.com/mathias-buder/water-level-measure)). For that I wanted to use, among other components, an Wmos D1 mini with the awesome Tasmota software on it. I had to use the special narrow FoV ultrasonic distance sensor [ME007-ULS](https://wiki.dfrobot.com/Water-proof%20Ultrasonic%20Sensor%20(ULS)%20%20SKU:%20SEN0300) and added it to the sensor driver library.

### CODE REVIEW

**Comment by arendst** [comment]
Thx.

To add features to Tasmota it's important to keep the code size a s small as possible. 

- Recently I standardized all ToF sensors from millimeter to centimeter. Pls adjust yours from meter to centimeter too eliminating the extra webserver GUI define.
- Pls remove the changes to .gitignore and platformio.ini.
- Pls find a way to remove stdlib.h and use Tasmota functions if needed.

File: tasmota/include/tasmota_template_legacy.h [hunk 1]
```diff
@@ -330,6 +330,8 @@ const uint16_t kGpioConvert[] PROGMEM = {
   AGPIO(GPIO_SBR_RX),         // Serial Bridge Serial interface
   AGPIO(GPIO_SR04_TRIG),      // SR04 Tri/TXgger pin
   AGPIO(GPIO_SR04_ECHO),      // SR04 Ech/RXo pin
+  AGPIO(GPIO_ME007_RX),       // ME007 Rx pin
```
Comment (arendst) [inline tasmota/include/tasmota_template_legacy.h:333]: Pls do not make changes to the legacy file. Not needed

**Comment by mathias-buder** [comment]
Hi Theo, thanks for you feeback. I've now made all the changes, let me know if anything else is needed.

File: tasmota/tasmota_xsns_sensor/xsns_23_me007.ino [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+#include <TasmotaSerial.h>
+
+/*********************************************************************************************/
+/* Defines */
+/*********************************************************************************************/
+#define XSNS_23                            23
+
+#define ME007_VERSION                      "1.0.0"                                              /**< Driver version X.Y.Z: X:Major, Y: Minor, Z: Patch */
+
+#define ME007_DEBUG_MSG_TAG                "ME007: "
```
Comment (arendst) [inline tasmota/tasmota_xsns_sensor/xsns_23_me007.ino:41]: Pls change to a three character tag like "ME7: "

File: tasmota/my_user_config.h [hunk 3]
```diff
@@ -782,8 +782,11 @@
 //#define USE_NOVA_SDS                             // Add support for SDS011 and SDS021 particle concentration sensor (+1k5 code)
   #define STARTING_OFFSET      30                // Turn on NovaSDS XX-seconds before tele_period is reached
 //#define USE_HPMA                                 // Add support for Honeywell HPMA115S0 particle concentration sensor (+1k4)
-//#define USE_SR04                                 // Add support for HC-SR04 ultrasonic devices (+1k code)
-  #define SR04_MAX_SENSOR_DISTANCE  500          // Set sensor max detection distance
+#define USE_SR04                                   // Add support for HC-SR04 ultrasonic devices (+1k code)
```
Comment (arendst) [inline tasmota/my_user_config.h:785]: As the current binaries are reaching max size pls disable both USE_SR04 (as was the case) and USE_ME007.

File: tasmota/tasmota_xsns_sensor/xsns_23_me007.ino (hunk 2 above)
Comment (mathias-buder) [reply tasmota/tasmota_xsns_sensor/xsns_23_me007.ino]: Ok, done!

File: tasmota/my_user_config.h (hunk 3 above)
Comment (mathias-buder) [reply tasmota/my_user_config.h]: Ok, done!

**Pull request merged**